Holovibes/sources/API.cc

```diff
@@ -733,61 +712,47 @@ void set_x_y(uint x, uint y)
 {
     if (get_compute_mode() == Computation::Raw || UserInterfaceDescriptor::instance().import_type_ == ImportType::None)
         return;
-    auto x_ = Holovibes::instance().get_setting<settings::X>().value;
-    if (x < Holovibes::instance().get_input_queue()->get_fd().width)
-    {
-        x_.start = x;
-        holovibes::Holovibes::instance().update_setting(holovibes::settings::X{x_});
-    }
 
-    auto y_ = Holovibes::instance().get_setting<settings::Y>().value;
-    if (y < Holovibes::instance().get_input_queue()->get_fd().width)
-    {
-        y_.start = y;
-        holovibes::Holovibes::instance().update_setting(holovibes::settings::Y{y_});
-    }
+    if (x < get_fd().width)
+        SET_SETTING(X, start, x);
+
+    if (y < get_fd().width)
+        SET_SETTING(Y, start, y);
+
     pipe_refresh();
 }
 
 void set_q_index(uint value)
 {
-    auto q = Holovibes::instance().get_setting<settings::Q>().value;
-    q.start = value;
-    holovibes::Holovibes::instance().update_setting(holovibes::settings::Q{q});
+    SET_SETTING(Q, start, value);
     pipe_refresh();
 }
 
 void set_q_accu_level(uint value)
 {
-    auto q = Holovibes::instance().get_setting<settings::Q>().value;
-    q.width = value;
-    holovibes::Holovibes::instance().update_setting(holovibes::settings::Q{q});
+    SET_SETTING(Q, width, value);
     pipe_refresh();
 }
 void set_p_index(uint value)
 {
-    auto p = Holovibes::instance().get_setting<settings::P>().value;
-    p.start = value;
-    holovibes::Holovibes::instance().update_setting(holovibes::settings::P{p});
+    SET_SETTING(P, start, value);
     pipe_refresh();
 }
 
 void set_p_accu_level(uint p_value)
 {
     UserInterfaceDescriptor::instance().raw_window.reset(nullptr);
 
-    auto p = Holovibes::instance().get_setting<settings::P>().value;
-    p.width = p_value;
-    holovibes::Holovibes::instance().update_setting(holovibes::settings::P{p});
+    SET_SETTING(P, width, p_value);
     pipe_refresh();
 }
 
 void set_composite_intervals(int composite_p_red, int composite_p_blue)
 {
-    holovibes::CompositeRGB rgb = Holovibes::instance().get_setting<settings::RGB>().value;
+    holovibes::CompositeRGB rgb = GET_SETTING(RGB);
     rgb.frame_index.min = composite_p_red;
     rgb.frame_index.max = composite_p_blue;
-    holovibes::Holovibes::instance().update_setting(holovibes::settings::RGB{rgb});
+    UPDATE_SETTING(RGB, rgb);
```

Comment:
It may be too complex and neesh, but is there a way to make your SET_SETTING macro to work with more than one path ?
If not no worries, im sure there are like 2 occurences where that would be useful

Reply:
No we need to create another macros in order to deal with 2 or more paths. Like SET_SETTING2. But i don't know if it will be overkill